src/main/java/Poirot.java

```diff
@@ -0,0 +1,204 @@
+                        break;
+                    default:
+                        System.out.println("Corrupted data: " + line);
+                }
+            }
+            scanner.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("No saved tasks found.");
+        }
+    }
+
+    public static void markTask(int index, boolean isDone) {
+        tasks[index].setDone(isDone);
+        saveTasksToFile();
+    }
+    public static void deleteTask(int index) {
+        Task removedTask = tasks[index];
+        for (int i = index; i < last_index - 1; i++) {
+            tasks[i] = tasks[i + 1];
+        }
+        tasks[last_index - 1] = null;
+        last_index--;
+        System.out.println(LINE);
+        System.out.println("Noted. I've removed this task:");
+        System.out.println("  " + removedTask);
+        System.out.println("Now you have " + last_index + " tasks in the list.");
+        System.out.println(LINE);
+    }
+    public static void main(String[] args) {
+        System.out.println(LINE);
+        System.out.println("Hello! I'm POIROT\n");
+        System.out.println("What can I do for you?");
+        System.out.println(LINE);
+
+        loadTasksFromFile();
+
+        boolean working = true;
+        Scanner scan = new Scanner(System.in);
+        while (working) {
+            String input = scan.nextLine();
+            String[] list_input = input.split(" ");
+            try {
+                switch (list_input[0]) {
+                    case "list":
+                        print(tasks);
+                        break;
+                    case "bye":
+                        working = false;
+                        break;
+                    case "mark":
+                        validateTaskNumber(list_input);
+                        int x = Integer.parseInt(list_input[1]) - 1;
+                        tasks[x].setDone(true);
+                        System.out.println(LINE);
+                        System.out.println("Nice! I've marked this task as done:\n");
+                        System.out.print("[" + tasks[x].getStatusIcon() + "] ");
+                        System.out.println(tasks[x].getDescription());
+                        System.out.println(LINE);
+                        saveTasksToFile(); // Save the task after marking it
+                        break;
+                    case "unmark":
+                        validateTaskNumber(list_input);
+                        int y = Integer.parseInt(list_input[1]) - 1;
+                        tasks[y].setDone(false);
+                        System.out.println(LINE);
+                        System.out.println("OK, I've marked this task as not done yet:\n");
+                        System.out.print("[" + tasks[y].getStatusIcon() + "] ");
+                        System.out.println(tasks[y].getDescription());
+                        System.out.println(LINE);
+                        saveTasksToFile();
```

Comment:
Consider saving the string formatted with the current `isDone` in a separate method inside `Task` class to reduce duplicate lines of code